src/main/java/org/apache/mesos/hdfs/executor/NameNodeExecutor.java

```diff
@@ -106,13 +113,53 @@ public void frameworkMessage(ExecutorDriver driver, byte[] msg) {
       } else {
         deleteFile(nameDir);
         nameDir.mkdirs();
-        runCommand(driver, nameNodeTask, "bin/hdfs-mesos-namenode " + messageStr);
-        startProcess(driver, nameNodeTask);
-        startProcess(driver, zkfcNodeTask);
+      }
+      runCommand(driver, nameNodeTask, "bin/hdfs-mesos-namenode " + messageStr);
+      startProcess(driver, nameNodeTask);
+      startProcess(driver, zkfcNodeTask);
+      driver.sendStatusUpdate(TaskStatus.newBuilder()
+          .setTaskId(nameNodeTask.taskInfo.getTaskId())
+          .setState(TaskState.TASK_RUNNING)
+          .setMessage(messageStr)
+          .build());
+    }
+  }
+
+  /**
+   * This class is designed to help launch the secondary namenode in its bootstrap phase. The node
+   * needs both name nodes to be available as DNS entries. In Mesos-DNS, this takes an uncertain
+   * amount of time, so we must schedule it asynchronously.
+   */
```

Comment:
The scheduler could be changed to format the nodes after it finds that they are running, but also it needs to check that they are discoverable which is a call on the scheduler. Why is it that you can't transition states once DNS is resolvable in the scheduler? The other option I see here is to have the methods MesosDns.journalNodesResolvable(conf, persistentState) called from the Executors instead of from the scheduler and use status messages to make the appropriate transitions. Either way will be fine, I just want to make sure we aren't duplicating code between the scheduler and executors unnecessarily. Let me know if I'm missing something, and we can also discuss tomorrow if needed.